Source/SwiftLintFramework/Rules/RuleConfigurations/FileHeaderConfiguration.swift

```diff
@@ -54,6 +58,8 @@ public struct FileHeaderConfiguration: RuleConfiguration, Equatable {
                                                     options: [.ignoreMetacharacters])
         } else if let requiredPattern = configuration["required_pattern"] {
             self.requiredPattern = requiredPattern
+            // For requiredPattern the regex might be recompiled in FileHeaderRule
```

Comment:
Ideally, we'd support `SWIFTLINT_CURRENT_FILENAME` in `forbiddenPattern` as well